### PR TITLE
wait for graphql server

### PR DESCRIPTION
**Pull request by liuliu-dev**

### CODE REVIEW

File: main/background.ts [hunk 1]
```diff
@@ -62,7 +62,36 @@ function createGraphqlSeverProcess() {
...
+  timeout: number = 30000,
+): Promise<void> {
+  const startTime = Date.now();
+  while (Date.now() - startTime < timeout) {
+    if (await isGraphQLServerReady(url)) {
+      return;
+    }
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+  throw new Error("GraphQL server did not become ready in time");
```
Comment (tbantle22) [inline main/background.ts]:
```suggestion
  throw new Error("Timed out starting GraphQL server");
```

File: main/background.ts [hunk 2]
```diff
@@ -62,7 +62,36 @@ function createGraphqlSeverProcess() {
...
+  try {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ query: "{ __typename }" }),
+    });
+    console.log(url, response);
+    return response.ok;
+  } catch (error) {
+    console.error("GraphQL server is not ready:", error);
```
Comment (tbantle22) [inline main/background.ts]:
```suggestion
    console.error("Error pinging GraphQL server:", error);
```

File: main/background.ts [hunk 3]
```diff
@@ -62,7 +62,36 @@ function createGraphqlSeverProcess() {
...
+
+async function waitForGraphQLServer(
+  url: string,
+  timeout: number = 30000,
+): Promise<void> {
+  const startTime = Date.now();
+  while (Date.now() - startTime < timeout) {
+    if (await isGraphQLServerReady(url)) {
+      return;
+    }
```
Comment (tbantle22) [inline main/background.ts]: Does the below work? I think it's bad practice to put an `await` in an if block like that
```suggestion
    const isReady = await isGraphQLServerReady(url)
    if (isReady) {
      return;
    }
```

**Review by tbantle22 — APPROVED** [review]
Seems like a reasonable approach to me. Both functions could use a comment explaining what they do

**Comment by liuliu-dev** [comment]
recreated the pull here: https://github.com/dolthub/dolt-workbench-desktop/pull/9
to get rid of the publish app changes, only includes the graphql server change

**Pull request closed without merging**